presto-hive/src/main/java/com/facebook/presto/twitter/hive/CachingHiveMetastoreRetryConfigurable.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.twitter.hive;
+
+import com.facebook.presto.hive.ForHiveMetastore;
+import com.facebook.presto.hive.HiveClientConfig;
+import com.facebook.presto.hive.HiveCluster;
+import com.facebook.presto.hive.RetryDriver;
+import com.facebook.presto.hive.metastore.CachingHiveMetastore;
+import com.facebook.presto.spi.PrestoException;
+import io.airlift.log.Logger;
+import io.airlift.units.Duration;
+
+import javax.annotation.concurrent.ThreadSafe;
+import javax.inject.Inject;
+
+import java.util.concurrent.ExecutorService;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * CachingHiveMetastoreRetryConfigurable is a subclass of CachingHiveMetastore.
+ * It added retry configuration and overrided the RetryDriver setup method
+ * retry().
+ * The class added the flexibility to set up retry configuration from a
+ * runtime injector in order to solve "Failed to connect hive metastore" issue.
```

Comment:
In discussions you mentioned that this code is only being added to debug this issue, implying that it shouldn't be permanently in place. Is that correct, or do we want it in the code even after that issues is resolved? If this is a temporary thing, please include that in the comments with a link to an issue to later remove it.